src/backend/distributed/commands/trigger.c

```diff
@@ -203,45 +242,480 @@ get_relation_trigger_oid_compat(HeapTuple heapTuple)
+ * command has been executed by standard process utility. This function errors out
+ * for unsupported commands or creates ddl job for supported ALTER TRIGGER DEPENDS
+ * ON commands.
+ */
+List *
+PostprocessAlterTriggerDependsStmt(Node *node, const char *commandString)
+{
+	AlterObjectDependsStmt *alterTriggerDependsStmt =
+		castNode(AlterObjectDependsStmt, node);
+	Assert(alterTriggerDependsStmt->objectType == OBJECT_TRIGGER);
+
+	RangeVar *relation = alterTriggerDependsStmt->relation;
+
+	bool missingOk = false;
+	LOCKMODE lockMode = AccessExclusiveLock;
+	Oid relationId = RangeVarGetRelid(relation, lockMode, missingOk);
+
+	if (!IsCitusTable(relationId))
+	{
+		return NIL;
+	}
+
+	ErrorIfUnsupportedTriggerCommand((Node *) alterTriggerDependsStmt, lockMode);
+
+	if (IsCitusLocalTable(relationId))
+	{
+		Value *triggerNameValue = GetAlterTriggerDependsTriggerNameValue(
+			alterTriggerDependsStmt);
+		return CitusLocalTableTriggerCommandDDLJob(relationId, strVal(triggerNameValue),
+												   commandString);
+	}
+
+	return NIL;
+}
+
+
+/*
+ * AlterTriggerDependsEventExtendNames extends relation name and trigger name
+ * with shardId, and sets schema name in given AlterObjectDependsStmt.
+ */
+void
+AlterTriggerDependsEventExtendNames(AlterObjectDependsStmt *alterTriggerDependsStmt,
+									char *schemaName, uint64 shardId)
+{
+	Assert(alterTriggerDependsStmt->objectType == OBJECT_TRIGGER);
+
+	RangeVar *relation = alterTriggerDependsStmt->relation;
+
+	char **relationName = &(relation->relname);
+	AppendShardIdToName(relationName, shardId);
+
+	Value *triggerNameValue = GetAlterTriggerDependsTriggerNameValue(
+		alterTriggerDependsStmt);
+	AppendShardIdToName(&strVal(triggerNameValue), shardId);
+
+	char **relationSchemaName = &(relation->schemaname);
+	SetSchemaNameIfNotExist(relationSchemaName, schemaName);
+}
+
+
+/*
+ * GetAlterTriggerDependsTriggerName returns Value object for the trigger
+ * name that given AlterObjectDependsStmt is executed for.
+ */
+static Value *
+GetAlterTriggerDependsTriggerNameValue(AlterObjectDependsStmt *alterTriggerDependsStmt)
+{
+	List *triggerObjectNameList = (List *) alterTriggerDependsStmt->object;
+	Value *triggerNameValue = llast(triggerObjectNameList);
+	return triggerNameValue;
+}
+
+
+/*
+ * PreprocessDropTriggerStmt is called before a DROP TRIGGER command has been
+ * executed by standard process utility. This function errors out for
+ * unsupported commands or creates ddl job for supported DROP TRIGGER commands.
+ */
+List *
+PreprocessDropTriggerStmt(Node *node, const char *commandString)
+{
+	DropStmt *dropTriggerStmt = castNode(DropStmt, node);
+	Assert(dropTriggerStmt->removeType == OBJECT_TRIGGER);
+
+	RangeVar *relation = GetDropTriggerStmtRelation(dropTriggerStmt);
+
+	bool missingOk = true;
+	LOCKMODE lockMode = AccessExclusiveLock;
+	Oid relationId = RangeVarGetRelid(relation, lockMode, missingOk);
+
+	if (!OidIsValid(relationId))
```

Comment:
Is the reason we don't have this check because this is a preprocess and postgres didn't do the validation yet?

Reply:
exactly